### PR TITLE
[AGENT-6475] Bump version of DRUM to latest in GPU envs

### PR DESCRIPTION
**Pull request by elatt**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
- Bump version of drum
- Pin transitive deps that are brought in

## Rationale
I want the status reporter bugfix included in the new envs.

### CODE REVIEW

File: public_dropin_gpu_environments/nim_llm/dr_requirements.txt [hunk 1]
```diff
@@ -12,54 +12,83 @@ anyio==4.6.2.post1
...
+cffi==1.17.1
 charset-normalizer==3.4.0
 click==8.1.7
+cryptography==43.0.3
 datarobot==3.5.2
-datarobot-drum==1.13.0
+datarobot-drum==1.14.1
 datarobot-mlops==10.2.0
 datarobot-mlops-connected-client==10.2.0
+datarobot-storage==0.0.0
```
Comment (elatt) [inline public_dropin_gpu_environments/nim_llm/dr_requirements.txt:30]: `datarobot-storage` pulls in all these new deps 🙃

File: public_dropin_gpu_environments/nim_llm/dr_requirements.txt [hunk 2]
```diff
@@ -71,8 +100,11 @@ tqdm==4.66.6
 trafaret==2.1.1
 typing-extensions==4.12.2
 tzdata==2024.2
-urllib3==2.2.3
+urllib3==1.26.20
```
Comment (elatt) [inline public_dropin_gpu_environments/nim_llm/dr_requirements.txt:103]: dr-storage is forcing us to downgrade this package 😢

File: public_dropin_gpu_environments/nim_llm/dr_requirements.txt (hunk 1 above)
Comment (klichukb) [reply public_dropin_gpu_environments/nim_llm/dr_requirements.txt:30]: darn.

File: public_dropin_gpu_environments/nim_llm/dr_requirements.txt (hunk 2 above)
Comment (klichukb) [reply public_dropin_gpu_environments/nim_llm/dr_requirements.txt:103]: Remember we had issues with urllib3 1 vs 2 incompat last year. No idea if this is any kind of threat but I'd say need to verify that this all still works with urllib 1

**Review by klichukb — APPROVED** [review]
We have some functional tests for other envs, not sure if we do for gpu, maybe makes sense to have something as well, to run smoke test (run drum & score something) in it.
For now I'd manually install this as bundle on staging and test if everything still works after a lot of dependency changes.

**Comment by elatt** [comment]
> We have some functional tests for other envs, not sure if we do for gpu, maybe makes sense to have something as well, to run smoke test (run drum & score something) in it.
> 
> For now I'd manually install this as bundle on staging and test if everything still works after a lot of dependency changes.

Yeah, we have functional tests but they need to be run on GPU instances. I reached out to harness and they gave me instructions on how to do it but I think the CI infra team needs to own supporting GPU instances.

I ran them manually and they passed which is why I feel ok about the urllib downgrade. I also scanned for CVEs we look good there too

**Review by alexromanyuk — APPROVED** [review]

**Pull request merged**